app/controllers/admin/variant_overrides_controller.rb

```diff
@@ -55,6 +56,20 @@ def load_data
         variant_override_enterprises_per_hub
 
       @inventory_items = InventoryItem.where(enterprise_id: @hubs)
+      @import_dates = inventory_import_dates.uniq.to_json
+    end
+
+    def inventory_import_dates
+      options = [{ id: '0', name: 'All' }]
+
+      import_dates = VariantOverride.
+        select('variant_overrides.import_date').
+        where('variant_overrides.hub_id IN (?)
+        AND variant_overrides.import_date IS NOT NULL', editable_enterprises.collect(&:id))
+
+      import_dates.uniq.collect(&:import_date).sort.reverse.map { |i| options.push(id: i.to_date, name: i.to_date.to_formatted_s(:long)) }
```

Comment:
can we do this in SQL, please, data-mangling at app level is one of our main performance root causes.